### DISCUSSION TITLE
eaf-browser error message: node_modules/darkreader/darkreader.js not found

### DISCUSSION THREAD

**Discussion by SoundingFunctions** [emacs-eaf/emacs-application-framework#1078] [Q&A] [answered]
hello,

just installed on macos.

eaf-open-browser gives me the following errors:
Buffer id '8905-9bfc-13f8-de70-132c-7656-221c' not exists, rebuild EAF buffer.

qt.webenginecontext: 

GL Type: core_profile
Surface Type: OpenGL
Surface Profile: CoreProfile
Surface Version: 4.1
QSG RHI Backend: OpenGL
Using Supported QSG Backend: yes
Using Software Dynamic GL: no
Using Multithreaded OpenGL: yes

Init Parameters:
  *  application-name eaf.py 
  *  browser-subprocess-path /Users/mstep/Library/Python/3.11/lib/python/site-packages/PyQt6/Qt6/lib/QtWebEngineCore.framework/Helpers/QtWebEngineProcess.app/Contents/MacOS/QtWebEngineProcess 
  *  disable-features ConsolidatedMovementXY,InstalledApp,BackgroundFetch,WebOTP,WebPayments,WebUSB,PictureInPicture 
  *  disable-speech-api  
  *  disable-web-security  
  *  enable-features NetworkServiceInProcess,TracingServiceInProcess 
  *  enable-gpu-rasterization  
  *  enable-native-gpu-memory-buffers  
  *  enable-threaded-compositing  
  *  ignore-gpu-blocklist  
  *  in-process-gpu  
  *  use-gl core_profile 

Traceback (most recent call last):
  File "/Users/mstep/.emacs.d/site-lisp/emacs-application-framework/core/utils.py", line 52, in on_signal_received
    self._func(obj, *args, **kwargs)
  File "/Users/mstep/.emacs.d/site-lisp/emacs-application-framework/eaf.py", line 155, in new_buffer
    self.create_buffer(buffer_id, url, module_path, arguments)
  File "/Users/mstep/.emacs.d/site-lisp/emacs-application-framework/eaf.py", line 173, in create_buffer
    app_buffer = module.AppBuffer(buffer_id, url, arguments)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mstep/.emacs.d/site-lisp/emacs-application-framework/app/browser/buffer.py", line 94, in __init__
    self.buffer_widget.init_dark_mode_js(__file__,
  File "/Users/mstep/.emacs.d/site-lisp/emacs-application-framework/core/webengine.py", line 736, in init_dark_mode_js
    self.dark_mode_js = open(os.path.join(os.path.dirname(module_path), "node_modules", "darkreader", "darkreader.js")).read()
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/mstep/.emacs.d/site-lisp/emacs-application-framework/app/browser/node_modules/darkreader/darkreader.js'

looking into:
.emacs.d/site-lisp/emacs-application-framework/app/browser

i see that there is no node-modules folder.

i found this
https://www.reddit.com/r/emacs/comments/v5a44d/unable_to_use_eafbrowser/ 

but it doesn-t help me as node_modules/darkreader/darkreader.js is missing from my system.

any hints?

**Comment by manateelazycat** [comment]
how did you install EAF？

**Reply by SoundingFunctions** [reply]
as stated here:
https://github.com/emacs-eaf/emacs-application-framework#2-installupdate-eaf-applications-and-dependencies

first:
git clone --depth=1 -b master https://github.com/emacs-eaf/emacs-application-framework.git ~/.emacs.d/site-lisp/emacs-application-framework/

then:
cd emacs-application-framework
chmod +x ./install-eaf.py
./install-eaf.py

**Comment by manateelazycat** [comment]
have error when install-eaf.py ?

**Reply by SoundingFunctions** [reply]
no. this is the output:

[EAF] ------------------------------------------
[EAF] Installing core dependencies
[EAF] Running pip3 install --user -U epc sexpdata tld lxml mac-app-frontmost PyQt6==6.3.1 PyQt6-Qt6==6.3.1 PyQt6-sip PyQt6-WebEngine==6.3.1 PyQt6-WebEngine-Qt6==6.3.1 @ /Users/mstep/.emacs.d/site-lisp/emacs-application-framework
Requirement already satisfied: epc in /Users/mstep/Library/Python/3.11/lib/python/site-packages (0.0.5)
Requirement already satisfied: sexpdata in /Users/mstep/Library/Python/3.11/lib/python/site-packages (1.0.0)
Requirement already satisfied: tld in /Users/mstep/Library/Python/3.11/lib/python/site-packages (0.13)
Requirement already satisfied: lxml in /Users/mstep/Library/Python/3.11/lib/python/site-packages (4.9.2)
Requirement already satisfied: mac-app-frontmost in /Users/mstep/Library/Python/3.11/lib/python/site-packages (2020.12.3)
Requirement already satisfied: PyQt6==6.3.1 in /Users/mstep/Library/Python/3.11/lib/python/site-packages (6.3.1)
Requirement already satisfied: PyQt6-Qt6==6.3.1 in /Users/mstep/Library/Python/3.11/lib/python/site-packages (6.3.1)
Requirement already satisfied: PyQt6-sip in /Users/mstep/Library/Python/3.11/lib/python/site-packages (13.4.1)
Requirement already satisfied: PyQt6-WebEngine==6.3.1 in /Users/mstep/Library/Python/3.11/lib/python/site-packages (6.3.1)
Requirement already satisfied: PyQt6-WebEngine-Qt6==6.3.1 in /Users/mstep/Library/Python/3.11/lib/python/site-packages (6.3.1)
[EAF] Finished installing core dependencies
[EAF] ------------------------------------------
[EAF] ------------------------------------------
[EAF] Installing application dependencies

[EAF] Updating pdf-viewer to newest version...
[EAF] Running git branch @ app/pdf-viewer
[EAF] Running git symbolic-ref HEAD @ app/pdf-viewer
[EAF] Running git pull origin master @ app/pdf-viewer
From https://github.com/emacs-eaf/eaf-pdf-viewer
 * branch            master     -> FETCH_HEAD
Already up to date.

[EAF] Updating browser to newest version...
[EAF] Running git branch @ app/browser
[EAF] Running git symbolic-ref HEAD @ app/browser
[EAF] Running git pull origin master @ app/browser
From https://github.com/emacs-eaf/eaf-browser
 * branch            master     -> FETCH_HEAD
Already up to date.

[EAF] Installing dependencies for the selected applications

[EAF] Please always ensure the following config are added to your init.el:
(require 'eaf-pdf-viewer)
(require 'eaf-browser)
[EAF] Installation SUCCESS!
[EAF] ------------------------------------------
[EAF] install-eaf.py finished.

**Comment by manateelazycat** [comment, marked as answer]
looks not did npm install or npm run build command for app directory

**Reply by SoundingFunctions** [reply]
thanks!

running:
brew install node

and then:
./install-eaf.py --force

did the trick.

**Comment by manateelazycat** [comment]
@MatthewZMD maybe we should check npm in install-eaf.py first? notify user if not found npm

**Reply by MatthewZMD** [reply]
We do have npm and nodejs as a system dependency.
